### ISSUE TITLE
Access via TOR

### ISSUE THREAD

**Issue by firechain** [prose/prose#982]
im using tor to access the internet. after the authentication with github i got this error message.

```
Error while loading data from Github. This might be a temporary issue. Please try again later.
```

**Comment by dereklieu** [comment]
I don't think using TOR would cause hiccups here. Just for kicks, can you try clearing your cookies for prose.io and re-authenticating with Github?

**Comment by dereklieu** [comment]
@firechain any updates here?

**Comment by firechain** [comment]
hi @dereklieu i tryed to clearing my cookies, TOR browser doesn't support cookies. maybe that's what cause the problems?

**Comment by dereklieu** [comment]
@firechain yeah, Prose sets a fairly long-lived oauth cookie that it uses to authenticate users between different users. Does your browser support `localstorage`?

**Comment by firechain** [comment]
@dereklieu, nope Tor's bundled browser doesn't support `localstorage`

**Comment by mikemorris** [comment]
Would it be possible to modify the auth flow to do something like the following?

1. Check to see if cookie exists.
2. If cookie exists, set auth key from cookie on global state.
3. If no cookie found, authenticate and set key on global state.
4. Attempt to set cookie from key on global state.

Thinking being that this flow could work while being tolerant of a failure to set the cookie - it would just slow everything down a bit by reauthenticating on full page reloads.

**Comment by dereklieu** [comment]
Yeah, it already sort of does something like the proposed workflow by setting a header on all requests using `$.ajaxSetup`. I think what's missing are guards against stray calls to `cookie.get('auth-header')` that just expect that to return something.
